packages/react/src/useActionForm.ts

```diff
@@ -67,6 +67,37 @@ const toDefaultValues = (modelApiIdentifier: string | undefined, data: any) => {
   return data;
 };
 
+const disambiguateDefaultValues = (data: any, initialData: any, action: any) => {
+  const initialKeys = Object.keys(initialData).filter((key) => !OmittedKeys.includes(key as OmittedKey));
+
+  const result = Object.fromEntries(
+    Object.entries(data).flatMap(([key, value]) => {
+      if (initialKeys.includes(key)) {
+        return [];
+      }
+
+      return [[key, value]];
+    })
+  );
+
+  const modelData = { ...data[action.modelApiIdentifier] };
+
+  for (const key of Object.keys(modelData)) {
+    const initialValue = initialData[key];
+    const flatValue = data[key];
+    const nestedValue = data[action.modelApiIdentifier]?.[key];
+
+    if (nestedValue && initialValue !== nestedValue) {
```

Comment:
This might need to be a `nestedValue !== undefined && initialValue !== nestedValue` so that `0` is a valid value. Or maybe even a `key in data[action.modelApiIdentifier]`

Reply:
:+1: added